### DISCUSSION TITLE
Visualization of EPOCH results

### DISCUSSION THREAD

**Discussion by SadaokiKojima** [epochpic/epoch#389] [General]
Dear Community Member.

I have a question about visualization of EPOCH results.
I am interested in visualizing laser accelerated fast electrons.

It is easy to visualize a map of the number density of electrons using either Visit or Python code.
(x-axis: x-space(um), y-axis: y-space(um), z-axis: electron number density)

However, without special post-processing, it is difficult to visualize fast electrons because bulk electrons dominate.

For example, to see fast electrons, the z-axis must be set to the number density of electrons with energies above 1 MeV.

Do you know how to do such a post-processing?

Best regards

Sadaoki

**Comment by Status-Mirror** [comment]
Hi Sadaoki,

Apologies for the delayed response - we're making an effort to address the backlog now. This sounds like a task for the subset block, which is documented here: https://epochpic.github.io/documentation/input_deck/input_deck_subset.html

You can get EPOCH to filter out particles which fulfil a set of criteria (in your case, a minimum relativistic gamma factor). This should reduce the effort in post-processing number densities.

Cheers,
Stuart
